### ISSUE TITLE
Add ignore: ["inside-function"] option to color-named

### ISSUE THREAD

**Issue by maraisr** [stylelint/stylelint#1736]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Error with rule `color-named` and `-s scss` cli option where by if i have it send to `"color-named": "never"` and in my scss file have `map-get($colours, red)` this is failing where it should be passing, becuase its pulling a "redish" colour from that map.

> Which rule, if any, is this issue related to?

`color-named`

> What CSS is needed to reproduce this issue?

e.g.

```css
.foo {
    color: map-get($colour, red);
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
   "color-named": "never"
  }
}
```

> Which version of stylelint are you using?

`7.0.3`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`stylelint $(find ./src/assets/sass -name *.scss) --colour -s scss`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

SCSS

> What did you expect to happen?

No warnings/errors to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.css
Unexpected named color "red"
```

**Comment by jeddy3** [comment]
@maraisr Thanks for the detailed report.

@stylelint/core I'm not sure there's anything we can (or want to) do about this, other than more clearly document that the rule simply checks for the named colours everywhere. For example, I'd argue that I'd expect this rule to checked for named colours within the `color()` function e.g. `color: color(red a(50%))`.

**Comment by maraisr** [comment]
@jeddy3 so then - my best bet would be to clone the rule, and extend the `stylelint-scss` plugin and hope for a merged PR?

**Comment by jeddy3** [comment]
It's labelled as discussion at the moment, so everyone usually throws in their thoughts and we try to pick the best path forward.

Perhaps there is something we can do in core that'll help in your case; a secondary `ignore` option i.e. `ignore: ["inside-function"]`?

**Comment by davidtheclark** [comment]
The idea to add an `ignore` option for `inside-function` or `inside-parentheses` sounds good to me.

**Comment by jeddy3** [comment]
> The idea to add an `ignore` option for `inside-function` or `inside-parentheses` sounds good to me.

Good stuff. `color-named` uses `postcss-value-parser` and so I think `"inside-function"` will be the easier of the two to implement. It's also an identifiable standard css construct. We've tended to use them, with the exception of `indentation` (where we have a `"inside-parens"` secondary option), for secondary options.

Anyone else have any thoughts on this issue, or shall I label it up as "help wanted"?

**Comment by jeddy3** [comment]
@maraisr Would you like to contribute this option? Step-by-step instructions are in the [developer guide](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-options-to-existing-rules).

**Comment by maraisr** [comment]
@jeddy3 yeah for sure - ill make this mod.

**Comment by maraisr** [comment]
You guys write way too advanced JavaScript for me, and the lack of documentation, i've found it next-to impossible to figure out how to get a simple prop added, and to test my file without running everything and so on... 

If you guys don't mind - can you please take a look at adding it? If not, feel free to close the issue, and ill put up with it.

**Comment by jeddy3** [comment]
> and the lack of documentation

We're trying to make contributing to stylelint as accessible as possible (as it's a large project and all help is greatly appreciated). Clearly there areas we can improve though. Can you help us improve the documentation by providing us with a few more details about where you got stuck?

Hopefully we can work together to improve the documentation and guidance, which will hopefully make it easier for you or another potential contributor to add this feature.

**Comment by gaidarenko** [comment]
Let me add this option. Label this issue as `wip`.

**Comment by m-allanson** [comment]
@gaidarenko 👍
